### ISSUE TITLE
Ability to add arbitrary configuration

### ISSUE THREAD

**Issue by bernardd** [bitnami/bitnami-docker-wordpress#114]
(There may already be a way to do this, but I haven't been able to find it).

It would be very useful to be able to add arbitrary configuration lines to the `wp-config.php` generated by the initialisation script. Our use case is this:
We're deploying a Wordpress container derived from the `bitnami/wordpress` one (thanks!) into a kubernetes cluster. The idea is that the container is deployed as-is, with all configuration etc in place, and just talks to an existing database, allowing us to bring down and up instances of it as required for scale, redundancy etc. The catch is that while it's easy to extend the image to add plugins we use, it's not easy to add config for those plugins to `wp-config.php` because that's generated on the first run rather than at the time the image is built.

What I'd propose is doing something like the dropins that Wordpress supports for modifying other behaviour by simply including a file if it exists:

```php
...
if ( file_exists(WP_CONTENT_DIR . '/custom-config.php') )
	require (WP_CONTENT_DIR . '/custom-config.php');

/* That's all, stop editing! Happy blogging. */
...
```

That way you keep all the nice easy setup that's already there, but allow for added flexibility when required.

Forgive me if this is a dumb idea or if there's already some easy way to achieve this - php and web stuff aren't my usual playground.

**Comment by juan131** [comment]
Hi @bernardd 

I think it could be interesting as long as this "extra" configuration has a specific format, well documented, etc. What do you think @jotamartos ?

As an alternative, you can create your own image based on this one but modifying the app-entrypoint.sh script: https://github.com/bitnami/bitnami-docker-wordpress/blob/master/4/rootfs/app-entrypoint.sh

You can add a command that adds your specific configuration to the `wp-config.php` file after WordPress is initialised.

**Comment by bernardd** [comment]
> I think it could be interesting as long as this "extra" configuration has a specific format, well documented, etc

What I'd imagined was that the extra config would simply be a snippit of code that was included in `wp-config.php`, which means the documentation for that config file would also cover the extra config (in addition to any required caveats regarding overwriting existing defines etc).

> You can add a command that adds your specific configuration to the wp-config.php file after WordPress is initialised.

I rejected that idea at the time, but for the life of me I can't now think why. I suspect it was because I thought I needed to get some extra stuff included just to get through the initialisation itself (connecting to the DB etc), and while that turns out not to be the case, it's still something to consider I think. For example it would provide a quick and easy way to work around issues such as https://github.com/bitnami/bitnami-docker-wordpress/issues/113 (say for people who already have a database with a given table prefix) without needing to wait for a patch.

**Comment by hadifarnoud** [comment]
I was just looking for a solution and came across this issue.

**Comment by juan131** [comment]
Hi @hadifarnoud 

What are you specifically trying to do? Are you trying to add custom configuration to **wp-config.php** ? Please provide as many details as possible so we can help you to find a workaround.

**Comment by hadifarnoud** [comment]
@juan131 after restoring backup of my wp blog, pod does not come up. see #136

**Comment by juan131** [comment]
Hi @hadifarnoud 

We're already discussing the changes needed to modify the `wp-config.php` at https://github.com/bitnami/bitnami-docker-wordpress/issues/131

Let's try to continue the discussion there to avoid confusion.

**Comment by hadifarnoud** [comment]
I'm still stuck with this issue. any way to edit wp-config file without having to build a custom image?

**Comment by marcosbc** [comment]
Hi @lwj5, we have just released tag [`5.7.1-debian-10-r21`](https://github.com/bitnami/bitnami-docker-wordpress#571-debian-10-r21) of the container images which implements a lot of new features. In particular it adds configuration specified via `WORDPRESS_EXTRA_WP_CONFIG_CONTENT` right before the "That's all, stop editing!" comment in the `wp-config.php` configuration file.

You can also specify `WORDPRESS_EXTRA_WP_CONFIG_CONTENT_FILE` and mount a file at that path, containing the additional file contents that you may want to add.
